### ISSUE TITLE
Night mode requires two requests

### ISSUE THREAD

**Issue by okonet** [cellcortex/homebridge-yeelighter#67]
Hey,

I was testing this plugin comparing it to https://github.com/vieira/homebridge-yeelight-wifi (I contributed the night mode feature there. See https://github.com/vieira/homebridge-yeelight-wifi/blob/master/bulbs/moonlight.js).

Unfortunately, the implementation this plugin is using (I was considering doing the same, TBH), doesn't work reliably with automations since it requires setting the value at least twice to toggle Night mode. I.e. when I simply set the brightness of the ceiling lamp to something less than 50, it blinks but returns to the daylight mode with the lower brightness. Same happens when going from `<50` to `>50` value. 

It maybe related to #47?

I'm on iOS `14.2.1`
Running Homebridge `1.1.7` (for some reason it doesn't work with the `1.3.0-beta40`)

**Comment by okonet** [comment]
Here are the logs I've got using `DEBUG` mode:

```
[04/01/2021, 15:57:40] [Yeelighter] [0x00000000075ebf14] transaction 3 took 0.159s [
  {
    id: 3,
    result: [
      'off', '2', '52',
      '',    '',  '4000',
      '',    '',  '',
      '',    '',  '',
      '0',   '0', ''
    ]
  }
]
[04/01/2021, 15:57:40] [Yeelighter] [0x00000000075ebf14] received update 3 after 0.16s: ["off","2","52","","","4000","","","","","","","0","0",""] []
[04/01/2021, 15:57:40] [Yeelighter] [0x00000000075ebf6e] transaction 3 took 0.162s [
  {
    id: 3,
    result: [
      'on', '2', '58',
      '',   '',  '3676',
      '',   '',  '',
      '',   '',  '',
      '26', '1', ''
    ]
  }
]
[04/01/2021, 15:57:40] [Yeelighter] [0x00000000075ebf6e] received update 3 after 0.162s: ["on","2","58","","","3676","","","","","","","26","1",""] []
[04/01/2021, 15:57:41] [Yeelighter] [0x000000000797a594] transaction 8 took 0.069s [ { id: 8, result: [ 'ok' ] } ]
[04/01/2021, 15:57:41] [Yeelighter] [0x000000000797a594] Moonlight off []
[04/01/2021, 15:57:41] [Yeelighter] [0x000000000797a594] set brightness [ 75 ]
[04/01/2021, 15:57:41] [Yeelighter] [0x000000000797a594] transaction 9 took 0.155s [ { id: 9, result: [ 'ok' ] } ]
[04/01/2021, 15:57:41] [Yeelighter] [0x000000000797a594] transaction 10 took 0.086s [ { id: 10, result: [ 'ok' ] } ]
[04/01/2021, 15:57:56] [Yeelighter] [0x000000000797a594] transaction 11 took 0.015s [ { id: 11, result: [ 'ok' ] } ]
[04/01/2021, 15:57:56] [Yeelighter] [0x000000000797a594] Moonlight on []
[04/01/2021, 15:57:56] [Yeelighter] [0x000000000797a594] set brightness [ 28 ]
[04/01/2021, 15:57:56] [Yeelighter] [0x000000000797a594] transaction 12 took 0.05s [ { id: 12, result: [ 'ok' ] } ]
[04/01/2021, 15:57:56] [Yeelighter] [0x000000000797a594] transaction 13 took 0.087s [ { id: 13, result: [ 'ok' ] } ]
[04/01/2021, 15:58:30] [Yeelighter] [0x000000000797a594] requesting attributes. Transactions: 5 []
[04/01/2021, 15:58:30] [Yeelighter] [0x000000000797a594] transaction 14 took 0.025s [
  {
    id: 14,
    result: [
      'on',  '2',    '56',
      '',    '',     '3690',
      'on',  '28',   '275',
      '100', '2703', '3',
      '0',   '0',    ''
```

**Comment by okonet** [comment]
Most weird part is if I use Homebridge UI and adjust the brightness there, it works as expected. 🤔

**Comment by okonet** [comment]
I added one light with additional logs so it's easier to track and this is what happens when I do the following: 

1. Turn the light on (it's in moonlight mode)
2. Increase the brightness (one tap in the iOS app)

```
[04/01/2021, 16:32:28] [Yeelighter] [Kids Test] sendCommand(81, set_power, ["on","smooth",500,5]) []
[04/01/2021, 16:32:28] [Yeelighter] [Kids Test] sent command 81: set_power []
[04/01/2021, 16:32:28] [Yeelighter] [Kids Test] received 81: OK []
[04/01/2021, 16:32:38] [Yeelighter] [Kids Test] sendCommand(82, set_power, ["on","sudden",0,1]) []
[04/01/2021, 16:32:38] [Yeelighter] [Kids Test] sent command 82: set_power []
[04/01/2021, 16:32:38] [Yeelighter] [Kids Test] sendCommand(83, set_power, ["on","smooth",500,5]) []
[04/01/2021, 16:32:38] [Yeelighter] [Kids Test] sent command 83: set_power []
[04/01/2021, 16:32:39] [Yeelighter] [Kids Test] received 82: OK []
[04/01/2021, 16:32:39] [Yeelighter] [Kids Test] set brightness [ 76 ]
[04/01/2021, 16:32:39] [Yeelighter] [Kids Test] sendCommand(84, set_bright, [52,"sudden",0]) []
[04/01/2021, 16:32:39] [Yeelighter] [Kids Test] sent command 84: set_bright []
[04/01/2021, 16:32:40] [Yeelighter] [Kids Test] received 83: OK []
[04/01/2021, 16:32:47] [Yeelighter] [Kids Test] sendCommand(85, get_prop, ["power","color_mode","bright","hue","sat","ct","bg_power","bg_bright","bg_hue","bg_sat","bg_ct","bg_lmode","nl_br","active_mode","name"]) []
[04/01/2021, 16:32:47] [Yeelighter] [Kids Test] sent command 85: get_prop []
[04/01/2021, 16:32:47] [Yeelighter] [Kids Test] temperature light updated {"power":true,"color_mode":2,"bright":44,"hue":0,"sat":0,"ct":3984,"bg_power":false,"bg_bright":0,"bg_hue":0,"bg_sat":0,"bg_ct":0,"bg_lmode":0,"nl_br":52,"active_mode":1,"name":""} []
```

The lamp turns on, when I update the brightness it lights up (moonlight mode turns off?) but then suddenly it goes back to moonlight mode again.

After a few seconds HomeKit app gets updated to the actual brightness (which is still under 50%)

**Comment by okonet** [comment]
Here is the same operation performed from the "Accessories" in the Homebridge UI:

```
[04/01/2021, 16:39:13] [Yeelighter] [Kids Test] sendCommand(97, set_power, ["on","smooth",500,1]) []
[04/01/2021, 16:39:13] [Yeelighter] [Kids Test] sent command 97: set_power []
[04/01/2021, 16:39:13] [Yeelighter] [Kids Test] received 97: OK []

[04/01/2021, 16:39:15] [Yeelighter] [Kids Test] sendCommand(98, set_power, ["on","sudden",0,5]) []
[04/01/2021, 16:39:15] [Yeelighter] [Kids Test] sent command 98: set_power []
[04/01/2021, 16:39:15] [Yeelighter] [Kids Test] received 98: OK []
[04/01/2021, 16:39:15] [Yeelighter] [Kids Test] set brightness [ 26 ]
[04/01/2021, 16:39:15] [Yeelighter] [Kids Test] sendCommand(99, set_bright, [52,"sudden",0]) []
[04/01/2021, 16:39:15] [Yeelighter] [Kids Test] sent command 99: set_bright []
[04/01/2021, 16:39:15] [Yeelighter] [Kids Test] received 99: OK []
```

**Comment by okonet** [comment]
So from what I can see it's the command 83 that's being triggered in the Home.app and not in the web UI:

```
[04/01/2021, 16:32:38] [Yeelighter] [Kids Test] sendCommand(83, set_power, ["on","smooth",500,5]) []
[04/01/2021, 16:32:38] [Yeelighter] [Kids Test] sent command 83: set_power []
...
[04/01/2021, 16:32:40] [Yeelighter] [Kids Test] received 83: OK []
```

**Comment by S474N** [comment]
I think, that this is problem directly in firmware. Please see my post on Yeelight Forum:
https://forum.yeelight.com/t/enable-moonlight-of-yeelight-ceiling-with-api/1182/14

**Comment by okonet** [comment]
Interesting that it works completely fine with the Switch accessory or via the Homebridge UI.

**Comment by S474N** [comment]
@okonet you don't have problems with 100% power on first turning on?

**Comment by okonet** [comment]
Not sure what problem you're describing but when I turn the light on it goes to the previous state. I don't have any problems with that. Also I can change the brightness without any issues.

**Comment by alondj99** [comment]
if you create a scene with the brightness that you need and run the automation on the scene instead of the accessory it works

**Comment by S474N** [comment]
> if you create a scene with the brightness that you need and run the automation on the scene instead of the accessory it works

And is it possible ot run this scene via Telnet/API?

**Comment by cellcortex** [comment]
I believe I have finally solved this in 2.4.20. The problem was that HomeKit sets the power to "on" (nearly) every time that the brightness is changed. This was racing with the mode change that was happening. Now there's some timers in place that should solve this.
